src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/CBApplicationBase.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    /// <summary>
+    /// The class for supporting multi-components based application model.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class CBApplicationBase : Application
+    {
+        private const string LogTag = "Tizen.Applications.CBApplicationBase";
+        private IList<BaseType> ComponentTypes = new List<BaseType>();
+        private Interop.CBApplication.CBAppLifecycleCallbacks _callbacks;
+        private bool _disposedValue = false;
+
+        /// <summary>
+        /// Initializes the CBApplicationBase class.
+        /// </summary>
+        /// <param name="typeInfo">The component type information.</param>
+        /// <exception cref="OutOfMemoryException">Thrown when failed because of out of memory.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public CBApplicationBase(IDictionary<Type, string> typeInfo)
+        {
+            _callbacks.OnInit = new Interop.CBApplication.CBAppInitCallback(OnInitNative);
+            _callbacks.OnFini = new Interop.CBApplication.CBAppFiniCallback(OnFiniNative);
+            _callbacks.OnRun = new Interop.CBApplication.CBAppRunCallback(OnRunNative);
+            _callbacks.OnExit = new Interop.CBApplication.CBAppExitCallback(OnExitNative);
+            _callbacks.OnCreate = new Interop.CBApplication.CBAppCreateCallback(OnCreateNative);
+            _callbacks.OnTerminate = new Interop.CBApplication.CBAppTerminateCallback(OnTerminateNative);
+
+            foreach (Type t in typeInfo.Keys)
+            {
+                RegisterComponent(t, typeInfo[t]);
+            }
+        }
+
+        /// <summary>
+        /// Registers a component.
+        /// </summary>
+        /// <param name="compType">Class type</param>
+        /// <param name="compId">Component ID</param>
+        /// <exception cref="InvalidOperationException">Thrown when component type is already added.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public void RegisterComponent(Type compType, string compId)
+        {
+            foreach (BaseType type in ComponentTypes)
+            {
+                if (compType.BaseType == type._classType)
+                    throw new InvalidOperationException("Already exist type");
+            }
+
+            if (compType.BaseType == typeof(BaseComponent))
```

Comment:
Do not directly compare with type info, becuase it will fail to sub classing case
```suggestion
            if (typeof(FrameComponent).IsAssignableFrom(compType))           
            else if (typeof(ServiceComponent).IsAssignableFrom(compType))
            else if (typeof(BaseComponent).IsAssignableFrom(compType))
```

Reply:
Done